test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/DataPartitionEnabledTests.cs

```diff
@@ -53,16 +53,31 @@ public async Task WhenRetrievingPartitions_TheServerShouldReturnAllPartitions()
     }
 
     [Fact]
-    [Trait("Category", "bvt-dp")]
     public async Task GivenDatasetWithNewPartitionName_WhenStoring_TheServerShouldReturnWithNewPartition()
+    {
+        var newPartition = new Partition(Partition.DefaultKey, TestUidGenerator.Generate());
+        await ValidateParititonCreation(newPartition);
+    }
+
+    [Fact]
+    [Trait("Category", "bvt-dp")]
+    public async Task GivenDatasetWithNewPartitionName_WhenStoringInparallel_TheServerShouldReturnWithNewPartition()
     {
         var newPartition = new Partition(Partition.DefaultKey, TestUidGenerator.Generate());
 
+        await Task.WhenAll(
```

Comment:
I've done manual testing of this by stepping through the SQL debugging in different processes, but I haven't automated such tests. Perhaps we can add many many partitions. However, we do not have a means of deleting partitions today, so tests like this will grow this infinitely.